frontend/src/components/Modal/CreateVirkaModal.tsx

```diff
@@ -28,8 +31,21 @@ interface CreateVirkaModalProps {
 const CreateVirkaModal: React.FC<CreateVirkaModalProps> = ({ open, handleClose }) => {
   const { t } = useTranslation();
   const dispatch = useAppDispatch();
+  const positionNames = useAppSelector(selectPositionNames);
+  const positionNamesLoading = useAppSelector(selectPositionNamesLoading);
+  const positionNameOptions = positionNames.map((option) => option.name);
+
+  useEffect(() => {
+    if (open) {
+      dispatch(fetchPositionNames());
+    }
+  }, [dispatch, open]);
 
   const onSubmit = async (values: Partial<Position>) => {
+    console.log(values);
+    const positionNameObj: PositionName = {
+      name: values.positionName?.toString() || '',
+    };
     try {
       const positionData: Position = {
```

Comment:
Could maybe add the OrgTreeId to Position model to match backend and prevent crashes, but can also be done in the next ticket